app/models/concerns/custom_field_addable.rb

```diff
@@ -83,7 +83,13 @@ def custom_value(field_identifier)
       field_identifier = parts[0]
       nested_attribute = parts[1]
     end
-    field = custom_field(field_identifier)
+
+    if field_identifier.is_a? CustomField
+      field = field_identifier
+    else
+      field = custom_field(field_identifier)
+    end
```

Comment:
This change makes the `field_identifier` variable name somewhat misleading. You might consider changing it to `field_or_id`.